### PR TITLE
Add command to display publish status for packages

### PR DESCRIPTION
**Pull request by imurchie**
Add a command (`--publish`, `--publish-status`, `-p`) to go through the packages and try to determine if there have been commits since they were last published. This is necessary in general, but in particular around publish-time for Appium as a whole. Without, we need to go through all the repositories by hand.

Also shift around argument handling, and silence npm as much as possible (thereby fixing #1).

### CODE REVIEW

File: lib/cli.js [hunk 1]
```diff
@@ -20,75 +23,124 @@ async function parseConfig (c) {
...
+      repo = clc.bgMagentaBright(repo);
+    } else {
+      if (!opts.showPublishedRepos) {
+        continue;
+      }
+    }
+
+    log(`${repo} (npm: ${pkg})`);
+    if (!_.isEmpty(commits)) {
+      for (const commit of commits) {
```
Comment (mykola-mokhnach) [inline lib/cli.js]: of (commits || [])

File: lib/packweb.js [hunk 2]
```diff
@@ -230,7 +247,7 @@ class PackWeb {
       try {
         res = await this.updateOwnersForPackage(p);
       } catch (e) {
-        if (e.message.indexOf("are not an owner") !== -1) {
+        if (e.message.includes("are not an owner")) {
```
Comment (mykola-mokhnach) [inline lib/packweb.js]:
```
if (!includes) {
  throw ...
}
res = ...
```

File: lib/packweb.js [hunk 3]
```diff
@@ -240,6 +257,58 @@ class PackWeb {
...
+      });
+    }
+
+    let repos = [];
+    for (const pkg of this.packages) {
+      let repo = await this.repoForPackage(pkg);
+      let owner = 'appium';
+      if (_.isNull(repo)) {
+        continue;
+      } else if (repo.includes('/')) {
```
Comment (mykola-mokhnach) [inline lib/packweb.js]: else if seems to be not necessary, since `continue` already breaks the flow

File: test/npm.js [hunk 4]
```diff
@@ -1,24 +1,32 @@
 /* eslint-disable promise/prefer-await-to-callbacks */
 
+
+const run = function run (args, map, cb) {
+  let res = map;
+  for (const arg of args) {
+    res = res[arg];
```
Comment (mykola-mokhnach) [inline test/npm.js]: what is the purpose of this reassigning?

File: lib/packweb.js [hunk 5]
```diff
@@ -240,6 +257,58 @@ class PackWeb {
...
+        ([owner, repo] = repo.split('/'));
+      }
+      let commits = await this.octokit.repos.getCommits({
+        owner,
+        repo,
+        per_page: commitDepth,
+      }).catch((err) => {
+        // the only error so far that should be skipped, it seems
+        if (!err.message.includes('Git Repository is empty')) {
+          return [err.message];
```
[the remark this answers is not in the document]
Comment (imurchie) [reply lib/packweb.js]: No need.

File: lib/packweb.js [hunk 6]
```diff
@@ -240,6 +257,58 @@ class PackWeb {
...
+        }
+        throw err;
+      });
+      commits = (commits.data || []).map(function (commit) {
+        return {
+          sha: commit.sha,
+          message: (commit.commit.message || '').split('\n')[0],
+          author: commit.author ? commit.author.login : '',
+          committer: commit.committer ? commit.committer.login : '',
```
[the remark this answers is not in the document]
Comment (imurchie) [reply lib/packweb.js]: That seems like a non-idiomatic way of doing the ternary operator. I'd rather it be as it is.

File: test/npm.js (hunk 4 above)
Comment (imurchie) [reply test/npm.js]: To crawl through the arguments and the data structure npm provides.

File: lib/packweb.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply lib/packweb.js]: this just covers the case when `login` property does not exist of is falsy

File: test/npm.js (hunk 4 above)
Comment (mykola-mokhnach) [reply test/npm.js]: It would be useful to add some comment there

**Pull request merged**